ZENKO-887 change to Installation/Install_guide.rst

wabernat opened the pull request:
This makes a one-line change to Install_guide.rst. Note, however, that I had to git add this file and directory to and paste in a copy of the file because the previous two versions are hung up in builds, so the history may look odd.

wabernat committed:
committing ZENKO-887 change to Installation/Install_guide.rst

-

ssalaues commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -0,0 +1,610 @@
+
+   Helm can now install applications on the Kubernetes cluster.
+
+3. Declare the ZooKeeper repository:
```
This section is no longer needed

ssalaues commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -0,0 +1,610 @@
+      $ helm repo add zenko-zookeeper https://scality.github.io/zenko-zookeeper/charts
+      "zenko-zookeeper" has been added to your repositories
+
+      $ helm repo add incubator http://storage.googleapis.com/kubernetes-charts-incubator
```
While these are no longer needed we need the following step `helm repo add scality https://scality.github.io/charts/`

wabernat committed:
Merge branch 'development/0.9' into bugfix/ZENKO-887_minimum_rootFS_req

wabernat committed:
Resolved conflicts, made revisions Salim requested.

Fixed a couple of formatting errors, one typo.

Removed ZENKO-633

Fixed more minor formatting issues.

updated issues per Salim's comments

removed:
autoscaling
lines 206-208
added note re NGINX ingress controller

Implementing Giacomo's changes.

Also added info for non-MetalK8s Helm installation.

picked a couple of nits,

added nodes 4 and 5 to cluster build intructions.

giacope reviewed (changes requested):
Not all the comments are requests.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+Cluster," below.
+
+The following section describes general cluster requirements, which are tested
+on Metal K8s. Because MetalK8s is designed to operate without support from
```
`MetalK8s`, not `Metal K8s`.

wabernat replied:
done.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+VMs) running CentOS_ 7.4 (The recommended mimimum for Zenko production service
+is five server nodes with three masters/etcds, but for testing and
+familiarization, three masters and three nodes is fine). You must have SSH
+access to these machines and they must have SSH access to each other. (You
```
No need for SSH access between nodes?

wabernat replied:
I don't understand the distinction you're making.

NicolasT replied:
> they must have SSH access to each other

That's not a requirement, and if it were it shouldn't be one.

wabernat replied:
OK, I talked through this. Fixed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+is five server nodes with three masters/etcds, but for testing and
+familiarization, three masters and three nodes is fine). You must have SSH
+access to these machines and they must have SSH access to each other. (You
+can copy SSH credentials from one machine to the next and log in once to
```
Never copy SSH private keys to other machines... There's no need to do it when deploying MetalK8s, and it's bad practice. Worst case you use agent forwarding.

wabernat replied:
fixed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+Sizing
+======
+
+Sizing for Metal K8s
```
`MetalK8s`

wabernat replied:
done.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+-  Storage (for the system)
+
+   -  20 GB for the root filesystem
+   -  16 GB for etcd
```
If we size `etcd` storage, we should document how to set up the system(s) to have `etcd` storage not on `/` (which is good practice in any case).

wabernat replied:
To whose expertise shall I appeal for this information?

wabernat replied:
Is this a show-stopper? I need to know either: a) that I can omit the etcd sizing requirement or b) how to configure the system to put etcd somewhere other than / .

If I don't hear back, I'm filing this under "nice to have" and moving the other changes along.

NicolasT replied:
I think this ought to be discussed with out TS folks. Right now, we don't have any recommendations for putting various types of data on different volumes, but maybe that should be the case.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+Proxies
+=======
+
+If you are behind a proxy, add the following lines to your local machine’s
```
What is 'your local machine'? It may need to be set on all machines in the cluster, no?

wabernat replied:
fixed

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+    no_proxy=localhost,127.0.0.1,10.*
+
+******************************
+Setting Up a Metal K8s Cluster
```
`MetalK8s`

wabernat replied:
done

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+
+.. code-block:: shell
+
+   $ git clone https://github.com/scality/metal-k8s
```
Using this, users would be installing some random in-development version of MetalK8s, rather than the one we want to have them deploy, so this is not the right approach to 'download' MetalK8s.

wabernat replied:
Where is the latest stable branch for clones to target?

NicolasT replied:
I'd recommend downloading the release files rather than pulling things from Git, for 'stable' installations, so https://github.com/scality/metalk8s/releases

wabernat replied:
Done.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+===================================
+
+Either log in to your primary machine (any of the nodes can be the primary)
+and clone the metal-k8s repo directly from GitHub:
```
`MetalK8s`

wabernat replied:
Done.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+
+   $ git clone https://github.com/scality/metal-k8s
+
+or put the metal-k8s project onto the machine using sftp:
```
`MetalK8s`

wabernat replied:
Done

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+Define an Inventory
+===================
+
+Each server must be configured in an inventory file that identifies servers
```
An inventory is *not* a *file*. It's a directory with various files, potentially in subdirectories.

wabernat replied:
fixed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+
+To specify the machines on which the Ansible_-based deployment system shall
+install MetalK8s, you must provide an *inventory*. This inventory contains a
+*hosts* file that lists all hosts in the cluster and *kube-node.yml*, a
```
`kube-node.yml` doesn't go next to the `hosts` file, but in a `group_vars` subdirectory of the inventory.

wabernat replied:
fixed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+
+   .. code-block:: yaml
+
+     metalk8s_lvm_default_vg: False
```
Why?

wabernat replied:
Salim suggested it. My build was not building. This seemed to move me a little closer to built.

wabernat replied:
removed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+   .. code-block:: yaml
+
+     metalk8s_lvm_default_vg: False
+     metalk8s_lvm_vgs: ['vg_metalk8s']
```
Shouldn't be required if `metalk8s_lvm_default_vg` were not set to `False`.

wabernat replied:
removed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+To install MetalK8s, you must issue commands from within a virtual shell.
+The following steps ensure you can access the virtual environment.
+
+1. Install python-virtualenv:
```
No need to install `python-virtualenv`?

wabernat replied:
removed.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+
+1. Make sure kubectl is installed on your local machine::
+
+   $ yum install kubectl
```
You get this in a `make shell` environment, which also ensures you get a version which works with the deployed cluster, which is definitely not guaranteed when installing from CentOS repositories.

wabernat replied:
deleted.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+   +-------------------------+---------------------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------+
+   | `Grafana`_              | Monitoring dashboards for cluster services              | http://localhost:8001/api/v1/namespaces/kube-ops/services/kube-prometheus-grafana:http/proxy/   |                                       |
+   +-------------------------+---------------------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------+
+   | `Cerebro`_              | An administration and monitoring console for            | http://localhost:8001/api/v1/namespaces/kube-ops/services/cerebro:http/proxy/                   | When accessing Cerebro, connect it to |
```
We pre-provision a config.

wabernat replied:
How would you have this reflected in the documentation? Is Cerebro preconfigured to elasticsearch:9200?

wabernat replied:
Following precedent in ZENKO-921 (Kibana, next comment) and deleting setup note.

NicolasT replied:
Yeah, there's an entry in a pre-populated list of 'known clusters' when you open Cerebro.

wabernat replied:
Done.

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+   |                         | Elasticsearch clusters                                  |                                                                                                 | http://elasticsearch:9200 to operate  |
+   |                         |                                                         |                                                                                                 | the MetalK8s Elasticsearch cluster.   |
+   +-------------------------+---------------------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------+
+   | `Kibana`_               | A search console for logs indexed in Elasticsearch      | http://localhost:8001/api/v1/namespaces/kube-ops/services/http:kibana:/proxy/                   | When accessing Kibana for the first   |
```
We pre-provision a config.

wabernat replied:
Fixed in Zenko-921

NicolasT commented on docs/Installation/Installation_Guide.rst:
```diff
@@ -1,579 +1,581 @@
+   |                         |                                                         |                                                                                                 | field name*.                          |
+   +-------------------------+---------------------------------------------------------+-------------------------------------------------------------------------------------------------+---------------------------------------+
+
+See :doc:`../architecture/cluster-services` for more about these services
```
Do these docs build?

wabernat replied:
Removed bad docs link. 

Can't comment about build other than to note that the merge request went through, and the linter for RST is quite permissive/nonexistent.

NicolasT replied:
This would only be found by actually building the docs. For MetalK8s, we're working on a step in CI which performs this, and as such makes sure any proposed change to the docs doesn't break its build.

smaffulli commented:
Folks, this commit (although merged) contains information that is not pertinent to Zenko IMO. How to install a MetalK8s cluster should be a link to MetalK8s docs. The instructions to install Zenko are also too MetalK8s-centric which is limiting.

I think we should revert this change. 

And before going any further, I would suggest to focus on agreeing on a common architecture for the whole Zenko documentation, setup the infrastructure to build it properly (take MetalK8s docs as an example to follow) and put the content we have in various formats/sources into one coherent information architecture.  Such conversation already started on https://github.com/scality/Zenko/pull/149

ssalaues commented:
@smaffulli I think the reason why we were okay with having this MetalK8s info on Zenko was because so many devs were having issues with specifically installing Zenko on MetalK8s. The docs for MetalK8s is great for just installing MetalK8s but we need a bit more configurations for Zenko on MetalK8s (mainly volume provisioning)

And if our devs were having pretty much the same issues with installs, likely others will run into the same issues as well.
